flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/ArcticCatalog.java

```diff
@@ -578,4 +606,29 @@ public InternalCatalogBuilder catalogBuilder() {
   public String amsCatalogName() {
     return internalCatalog.name();
   }
+
+  /**
+   * Check whether a list of partition values are valid based on the given list of partition keys.
+   *
+   * @param partitionSpec a partition spec.
+   * @param arcticPartitionSpec arcticPartitionSpec
+   * @param tablePath tablePath
+   * @throws PartitionSpecInvalidException thrown if any key in partitionSpec doesn't exist in
+   *     partitionKeys.
+   */
+  private void checkValidPartitionSpec(
+      CatalogPartitionSpec partitionSpec, PartitionSpec arcticPartitionSpec, ObjectPath tablePath)
+      throws PartitionSpecInvalidException {
+    for (PartitionField field : arcticPartitionSpec.fields()) {
+      if (!partitionSpec.getPartitionSpec().containsKey(field.name())) {
```

Comment:
I think we should check that `arcticPartitionSpec.fields()` contains all the `partitionSpec.getPartitionSpec().keySet()`
```suggestion
      List<String> partitionKeys = spec.fields().stream().map(PartitionField::name).collect(Collectors.toList());
      for (String key : partitionSpec.getPartitionSpec().keySet()) {
            if (!partitionKeys.contains(key)) {
                throw new PartitionSpecInvalidException(
                        getName(), partitionKeys, tablePath, partitionSpec);
            }
        }
```

Reply:
fixd.